manifests/server/puppetserver.pp

```diff
@@ -244,9 +245,15 @@
     content => template('puppet/server/puppetserver/conf.d/puppetserver.conf.erb'),
   }
 
-  file { "${server_puppetserver_dir}/conf.d/auth.conf":
-    ensure  => file,
-    content => template('puppet/server/puppetserver/conf.d/auth.conf.erb'),
+  if $server_auth_template {
+    file { "${server_puppetserver_dir}/conf.d/auth.conf":
+      ensure => file,
+      content => template($server_auth_template),
+    }
+    file { "${server_puppetserver_dir}/conf.d/auth.conf":
+      ensure  => file,
+      content => template('puppet/server/puppetserver/conf.d/auth.conf.erb'),
+    }
```

Comment:
I think you're missing an `else` but I'd prefer `pick($server_auth_template, 'puppet/server/puppetserver/conf.d/auth.conf.erb')`.

Reply:
sorry, that was super sloppy of me.